src/native/libs/configure.cmake

```
@@ -458,6 +458,18 @@ check_symbol_exists(
     sys/epoll.h
     HAVE_EPOLL)
 
+check_symbol_exists(
+    gethostname
+    unistd.h
+    HAVE_GETHOSTNAME)
+
+check_symbol_exists(
+    getnameinfo
+    netdb.h
+    HAVE_GETNAMEINFO)
+
+check_struct_has_member("struct sockaddr_un" sun_path "sys/types.h;syn/un.h" HAVE_SOCKADDR_UN_SUN_PATH)
```

I think the issue is just that you have a typo :)

```suggestion
check_struct_has_member("struct sockaddr_un" sun_path "sys/types.h;sys/un.h" HAVE_SOCKADDR_UN_SUN_PATH)
```

the define should also be called `HAVE_SOCKADDR_UN` for consistency

---

> the define should also be called `HAVE_SOCKADDR_UN` for consistency

No, the struct is there, but it doesn't have all fields.